ocean/src/social.rs

```diff
@@ -31,14 +57,33 @@ impl SocialNetwork {
     * Result enum
     */
     pub fn remove_friend(&mut self, crab_id: String, friend_id: String) -> Result<(), String> {
-        unimplemented!();
+        match self.friendships.get_mut(&crab_id) {
+            Some(friends) => {
+                if friends.contains(&friend_id) {
+                    friends.retain(|f| f != &friend_id);
+                    Ok(())  // Friend was found and removed
+                } else {
+                    Err("Friend not found in list".to_string())  // Friend was not in the list
+                }
+            },
+            None => Err("Crab ID does not exist or has no friends".to_string())  // Crab ID does not exist or has no friends
+        }
     }
 
     /*
     Returns True if crab2 is a friend of crab1, else False.
     */
-    pub fn is_friend(&self, crab_id: String, friend_id: String) -> bool {
```

Comment:
What do you think about changing `crab_id` to a more generic name like `id` or `entity_id`? I think that could be nice to have in case some other beach has seals that also form a social network.